dash/include/dash/memory/GlobStaticMem.h

```diff
@@ -523,9 +524,10 @@ class GlobStaticMem
     DASH_LOG_TRACE_VAR("GlobStaticMem.at", lunit);
     // Apply global unit to global pointer:
     dart_gptr_setunit(&gptr, lunit);
+    // increment locally only
+    gptr.addr_or_offs.offset += local_index * sizeof(value_type);
     // Apply local offset to global pointer:
     pointer res_gptr(*this, gptr);
-    res_gptr += local_index;
```

Comment:
Why does this not work? Shouldn't it be the same as incrementing the offset if the `dart_gptr_t` by `local_index * sizeof(value_type)`?

Reply:
The difference is that the dart version only increments locally while the dash version first checks if the increment is past the local address range. As the local size of the memory on unit 1 is 96 (vs 144, unit 0) the dash version belives that a increment of 128 is a pointer to index 32 on the next unit. The actual bug are the capacity issues as stated in the linked issue.

Reply:
Ahh, gotcha! I agree that it seems more efficient to operate on the `dart_gptr_t` directly, anyway. Thanks for fixing this!